test: less flaky date control test

netchampfaris opened the pull request:
- Assert the actual value of the field rather than the formatted value
- Always start a new test from a clean state, currently the date test assumes the state created by the previous test
- No need to create an entire doctype just to test a single field, can be done in a dialog

netchampfaris committed:
test: less flaky date control test

netchampfaris committed:
style: semicolons

surajshetty3416 commented:
@netchampfaris date control has some code due to which date control can act differently in form view. So I think it is better to keep a case where date control is validated in form view as well (or remove `frm` dependent code?).

netchampfaris commented:
@surajshetty3416 Don't see any `frm` specific code except this which is only used for positioning

https://github.com/frappe/frappe/blob/b2fc959307c7c79f5584625569d5aed04133ba13/frappe/public/js/frappe/form/controls/date.js#L98-L100

surajshetty3416 commented:
> @surajshetty3416 Don't see any `frm` specific code except this which is only used for positioning

ideally, shouldn't that be covered as well to verify if that part of the code does not have any side-effect in rendering the control?

netchampfaris commented:
@surajshetty3416 Diminishing returns, not worth it.

netchampfaris committed:
fix: check dates in frappe realm

new Date in Cypress context and new Date in frappe context differs sometimes in CI

netchampfaris committed:
chore: pretty logging for dialog command
